base/abstractarray.jl

```diff
@@ -1048,6 +1048,50 @@ function _setindex!(::IndexCartesian, A::AbstractArray, v, I::Vararg{Int,M}) whe
     r
 end
 
+## rudimentary aliasing detection ##
+"""
+    unalias(dest, A)
+
+Return either `A` or a copy of `A`, in a rough effort to prevent modifications to `dest` from
+affecting the returned object. No guarantees are provided, and each custom array must
+opt-into aliasing detection by overloading this method by specializing on the second argument.
+
+This function must return an object of exactly the same type as `A` for performance and type-stability.
+
+See also [`mightalias`](@ref) and [`dataids`](@ref).
+"""
+unalias(dest, A) = A
+unalias(dest, A::Array) = mightalias(dest, A) ? copy(A) : A
+@inline unalias(dest, As::Tuple) = (unalias(dest, As[1]), unalias(dest, tail(As))...)
```

Comment:
I think this would make more sense as a varargs function; otherwise it seems like you're unaliasing the tuple itself (which is somewhat array-like, so could be confusing).

Reply:
Yeah, that was Milan's comment, too.  I can make that change.  https://github.com/JuliaLang/julia/pull/25890#discussion_r166006058